### PR TITLE
Use constants for dataflow/stream property keys

### LINKED ISSUE #1297: Create constant class for stream and metric keys
`StreamDeploymentController` for metrics work in #1291 is adding keys like `spring.cloud.stream.metrics.key`. 

Comment from @ilayaperumalg related to that
```
All the spring.cloud.stream.metrics properties' keys could probably be defined in BindingPropertyKeys Class. Perhaps, we could rename the Class BindingPropertyKeys to SpringCloudStreamPropertyKeys which would also lead a way to create SpringCloudDataFlowPropertyKeys for other property keys that start with prefix spring.cloud.dataflow.
```
Creating ticket here is it's a bit complex issue to get handled in a PR itself.

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Define constant prefix/names wherever possible
 - Update `spring.cloud.dataflow.` prefix in other places in the codebase as well

Resolves #1297

### CODE REVIEW

File: spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/DataFlowPropertyKeys.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ */
+
+package org.springframework.cloud.dataflow.core;
+
+/**
+ * Spring Cloud Data Flow property keys.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public class DataFlowPropertyKeys {
```
Comment (markfisher) [inline spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/DataFlowPropertyKeys.java]: can the `PREFIX` vars here all be `private`?

Comment (ilayaperumalg) [inline spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/DataFlowPropertyKeys.java]: yeah, but we wanted to use this `PREFIX` in other places especially the `ConfigurationProperties` classes.

Comment (markfisher) [inline spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/DataFlowPropertyKeys.java]: but I believe `STREAM_APP_PREFIX` and `STREAM_PREFIX` can be private

Comment (ilayaperumalg) [inline spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/DataFlowPropertyKeys.java]: yeah, sure.

**Comment by jvalkeal** [comment]
squashed/merged per https://github.com/spring-cloud/spring-cloud-dataflow/commit/9983bc7c2dc11e77b7320f7c37a59eaf5b542682

**Pull request closed without merging**